### PR TITLE
fix: fix minor issues found in audit

### PR DESCRIPTION
**Pull request by adu-web3**
## Description

There are some minor issues identified in the audit by certik, we could address them in one PR as they are not complex or hard to fix

- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=BEN-04
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=ENB-01
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=OAS-02
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=EGE-03
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=ECE-02
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=ECE-01
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=BEN-03
- [x] https://skyharbor.certik.com/report/04d25ca5-49da-4609-99cf-e59242a8914e?findingIndex=BEN-01

## Summary by CodeRabbit

- **New Features**
	- Introduced a function to withdraw non-Beacon Chain ETH from the ExoCapsule, enhancing user capabilities.
	- Added stricter checks for non-zero Ether transactions across multiple functions to improve security.

- **Bug Fixes**
	- Updated withdrawal logic to clarify user context and prevent unintended Ether transfers.

- **Documentation**
	- Enhanced function documentation for clarity on ETH withdrawal processes and parameters.

- **Tests**
	- Expanded test coverage for Ether transactions to ensure correct handling and reversion scenarios.

### CODE REVIEW

File: src/interfaces/IExoCapsule.sol [hunk 1]
```diff
@@ -45,12 +45,17 @@ interface IExoCapsule {
...
+    /// @notice Allows the owner to withdraw the specified unlocked staked ETH to the recipient.
     /// @dev The amount must be available in the withdrawable balance.
     /// @param amount The amount to withdraw.
     /// @param recipient The recipient address.
     function withdraw(uint256 amount, address payable recipient) external;
 
+    /// @notice Withdraws the nonBeaconChainETHBalance
+    /// @param recipient The payable destination address to which the ETH are sent.
+    /// @param amountToWithdraw The amount to withdraw.
+    function withdrawNonBeaconChainETHBalance(address payable recipient, uint256 amountToWithdraw) external;
```
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/interfaces/IExoCapsule.sol:57]: sure

File: src/core/NativeRestakingController.sol [hunk 2]
```diff
@@ -130,4 +130,18 @@ abstract contract NativeRestakingController is
...
+    /// @param amountToWithdraw The amount to withdraw.
+    function withdrawNonBeaconChainETHFromCapsule(address payable recipient, uint256 amountToWithdraw)
+        external
+        whenNotPaused
+        nonReentrant
+        nativeRestakingEnabled
+    {
+        IExoCapsule capsule = _getCapsule(msg.sender);
+        capsule.withdrawNonBeaconChainETHBalance(recipient, amountToWithdraw);
+    }
```
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/core/NativeRestakingController.sol:145]: already added

File: src/interfaces/IExoCapsule.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (adu-web3) [reply src/interfaces/IExoCapsule.sol:57]: yes

**Review by bwhour — APPROVED** [review]

File: src/core/ExoCapsule.sol [hunk 3]
```diff
@@ -266,11 +268,14 @@ contract ExoCapsule is ReentrancyGuardUpgradeable, ExoCapsuleStorage, IExoCapsul
...
 
     /// @notice Withdraws the nonBeaconChainETHBalance
-    /// @dev This function must be called through the gateway. @param amount must be greater than
+    /// @dev This function must be called through the gateway. @param amountToWithdraw can not be greater than
     /// the available nonBeaconChainETHBalance.
-    /// @param recipient The destination address to which the ETH are sent.
+    /// @param recipient The payable destination address to which the ETH are sent.
     /// @param amountToWithdraw The amount to withdraw.
-    function withdrawNonBeaconChainETHBalance(address recipient, uint256 amountToWithdraw) external onlyGateway {
+    function withdrawNonBeaconChainETHBalance(address payable recipient, uint256 amountToWithdraw)
```
Comment (MaxMustermann2) [inline src/core/ExoCapsule.sol:275]: Not sure about this but should the parameter for _sendETH be made payable as well?

**Review by MaxMustermann2 — APPROVED** [review]

File: src/core/ExoCapsule.sol (hunk 3 above)
Comment (adu-web3) [reply src/core/ExoCapsule.sol:275]: fixed in commit [edd35c9](https://github.com/ExocoreNetwork/exocore-contracts/pull/82/commits/edd35c9939fb8afd0157382adf0d3eeff3cf618a)

**Pull request merged**